tsl/src/planner.c

```diff
@@ -231,6 +238,46 @@ tsl_create_distributed_insert_path(PlannerInfo *root, ModifyTablePath *mtpath, I
 
 			table_close(rel, AccessShareLock);
 		}
+		else /* check if it's a INSERT ... SELECT case involving dist hypertables */
+		{
+			RangeTblEntry *select_rte = NULL;
+			ListCell *l;
+			RangeTblEntry *rte = planner_rt_fetch(hypertable_rti, root);
+			bool distributed = false;
+
+			/* if src hypertable is distributed then only further checks are needed */
+			if (ts_rte_is_hypertable(rte, &distributed) && distributed)
+			{
+				/* check if it's indeed INSERT .. SELECT */
+				foreach (l, root->parse->rtable)
+				{
+					rte = (RangeTblEntry *) lfirst(l);
+
+					if (rte->rtekind == RTE_SUBQUERY)
```

Comment:
Is it not pulled up for INSERTs? I'd expect to see an RTE_RELATION here.

Reply:
Not it's not pulled up. You see an RTE for the table being inserted into and then an RTE for the subquery followed by RTE entries for the subquery rels